### PR TITLE
Create repository factory for apptoken

### PR DESCRIPTION
**Pull request by uruemu**

### CODE REVIEW

**Review by cameronperera — APPROVED** [review]
LGTM

**Comment by jimlambrt** [comment]
Can you rebase so it only includes the new commits?

File: internal/apptoken/options.go [hunk 1]
```diff
@@ -45,3 +46,12 @@ func WithDescription(_ context.Context, desc string) Option {
 		return nil
 	}
 }
+
+// WithLimit provides an option to provide a limit.  Intentionally allowing
+// negative integers.
+func WithLimit(_ context.Context, limit int) Option {
```
Comment (jimlambrt) [inline internal/apptoken/options.go:52]: unit test?

**Review by jimlambrt — CHANGES_REQUESTED** [review]
Looks great, one minor bit.

File: internal/apptoken/options.go (hunk 1 above)
Comment (uruemu) [reply internal/apptoken/options.go:52]: Added a test!

**Pull request merged**